File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DateTimePicker/Features/LangaugesPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;
using System.Globalization;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.DateTimePicker.Features
{
    public partial class LangaugesPage : UserControl
    {
        private CultureInfo _currentUICulture;
        public LangaugesPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles Load event of the example' UserControl.
        /// </summary>
        private void LangaugesPage_Load(object sender, EventArgs e)
        {
            _currentUICulture = Context.CurrentUICulture;
            // Fill up the ConboBox with support languages.
            this.mobjLanguagesComboBox.DataSource = Enum.GetValues(typeof(SupportLanguages));
            
        }

        /// <summary>
        /// Handles SelectedIndexChanged event of the ComboBox that contains 
        /// support languages. Sets selected languages for the current page.
        /// </summary>
        private void mobjLanguagesComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            // Set selected languages for the current page.
            SupportLanguages selectedLanguage = (SupportLanguages)this.mobjLanguagesComboBox.SelectedItem;
            switch (selectedLanguage)
            {
                case SupportLanguages.English:
                    Context.CurrentUICulture = new CultureInfo("en-US", false);
                    break;
                case SupportLanguages.French:
                    Context.CurrentUICulture = new CultureInfo("fr-FR", false);
                    break;
                case SupportLanguages.German:
                    Context.CurrentUICulture = new CultureInfo("de-DE", false);
                    break;
                case SupportLanguages.Hebrew:
                    Context.CurrentUICulture = new CultureInfo("he-IL", false);
                    break;
                default:
                    Context.CurrentUICulture = new CultureInfo("en-US", false);
                    break;
            }
            
        }
        /// <summary>
        /// Represents support languages
        /// </summary>
        private enum SupportLanguages
        {
            English,
            French,
            German,
            Hebrew
        }

        /// <summary>
        /// Handles VisibleChanged event of the example' UserControl.
        /// Sets default globalization settings for the project.
        /// </summary>
        private void LangaugesPage_VisibleChanged(object sender, EventArgs e)
        {
            if (_currentUICulture != null)
            {
                Context.CurrentUICulture = _currentUICulture;
            }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DateTimePicker/Features/LangaugesPage.vb
Imports System.Globalization
Namespace CompanionKit.Controls.DateTimePicker.Features

    Public Class LangaugesPage

        Private _currentUICulture As CultureInfo


        Public Sub New()

            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()
        End Sub

        ''' <summary>
        ''' Handles Load event of the example' UserControl.
        ''' </summary>
        Private Sub LangaugesPage_Load(sender As Object, e As EventArgs) Handles Me.Load
            _currentUICulture = Context.CurrentUICulture
            ' Fill up the ConboBox with support languages.
            Me.mobjLanguagesComboBox.DataSource = [Enum].GetValues(GetType(SupportLanguages))

        End Sub

        ''' <summary>
        ''' Handles SelectedIndexChanged event of the ComboBox that contains
        ''' support languages. Sets selected languages for the current page.
        ''' </summary>
        Private Sub mobjLanguagesComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjLanguagesComboBox.SelectedIndexChanged
            ' Set selected languages for the current page.
            Dim selectedLanguage As SupportLanguages = CType(Me.mobjLanguagesComboBox.SelectedItem, SupportLanguages)
            Select Case selectedLanguage
                Case SupportLanguages.English
                    Context.CurrentUICulture = New CultureInfo("en-US", False)
                    Exit Select
                Case SupportLanguages.French
                    Context.CurrentUICulture = New CultureInfo("fr-FR", False)
                    Exit Select
                Case SupportLanguages.German
                    Context.CurrentUICulture = New CultureInfo("de-DE", False)
                    Exit Select
                Case SupportLanguages.Hebrew
                    Context.CurrentUICulture = New CultureInfo("he-IL", False)
                    Exit Select
                Case Else
                    Context.CurrentUICulture = New CultureInfo("en-US", False)
                    Exit Select
            End Select

        End Sub
        ''' <summary>
        ''' Represents support languages
        ''' </summary>
        Private Enum SupportLanguages
            English
            French
            German
            Hebrew
        End Enum

        ''' <summary>
        ''' Handles VisibleChanged event of the example' UserControl.
        ''' Sets default globalization settings for the project.
        ''' </summary>
        Private Sub LangaugesPage_VisibleChanged(sender As Object, e As EventArgs) Handles Me.VisibleChanged
            If _currentUICulture IsNot Nothing Then
                Context.CurrentUICulture = _currentUICulture
            End If
        End Sub
    End Class

End Namespace